### PR TITLE
Stop Generation works with messages stuck in created

### PR DESCRIPTION
**Pull request by PopDaph**
Eng runner card: https://github.com/dust-tt/tasks/issues/102

- When clicking on the "Stop Generation" button, on top of emitting a Redis key (for which I added an expiration date) that will be managed by `runGeneration` to stop the generation, we immediately set the status of the message to `cancelled`. After this action, we mutate the conversation.

On `AgentMessage`:
- On the UI we consider that a message is streaming if both `streamedAgentMessage` and `message` have a status to created.
- `agentMessageToRender`: if the message is in cancelled but has no content yet we return it with the content already streamed (to handle the case when the backend has not already stopped the generation and saved the content). 

WDYT?

### CODE REVIEW

**Review by spolu — COMMENTED** [review]
Left a few questions

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 1]
```diff
@@ -163,8 +150,11 @@ export function AgentMessage({
     switch (message.status) {
       case "succeeded":
       case "failed":
-      case "cancelled":
         return message;
+      case "cancelled":
+        return message.content
```
[the remark this answers is not in the document]
Comment (PopDaph) [reply front/components/assistant/conversation/AgentMessage.tsx:155]: When you refresh yes but between the moment you clicked and we set the status to cancel and the moment the generation is stopped and the message is set with the content that was stream, it causes a glitch where we display empty and then the content.

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 2]
```diff
@@ -61,21 +61,8 @@ export function AgentMessage({
...
-      case "created":
-        return true;
-
-      default:
-        ((status: never) => {
-          throw new Error(`Unknown status: ${status}`);
-        })(streamedAgentMessage.status);
-    }
-  })();
+  const shouldStream =
```
[the remark this answers is not in the document]
Comment (PopDaph) [reply front/components/assistant/conversation/AgentMessage.tsx:64]: Cause when we set message.status to cancelled we want to stop the generation even if streamedAgentStatus is still in created 🙏🏻

[the remark this answers is not in the document]
Comment (PopDaph) [reply front/components/assistant/conversation/AgentMessage.tsx:64]: We still have a throw on `buildEventSourceURL` if we received an event with a non expected type

**Review by spolu — DISMISSED** [review]

File: front/lib/api/assistant/pubsub.ts [hunk 3]
```diff
@@ -306,9 +307,34 @@ export async function cancelMessageGenerationEvent(
   messageIds: string[]
 ): Promise<void> {
   const redis = await redisClient();
-  messageIds.forEach(async (messageId) => {
-    await redis.set(`assistant:generation:cancelled:${messageId}`, 1);
+
+  const tasks = messageIds.map((messageId) => {
```
Comment (lasryaric) [inline front/lib/api/assistant/pubsub.ts:311]: as discussed, try / catch / finally { await redis.quit()}

**Review by lasryaric — CHANGES_REQUESTED** [review]

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 4]
```diff
@@ -163,8 +167,11 @@ export function AgentMessage({
     switch (message.status) {
       case "succeeded":
       case "failed":
-      case "cancelled":
         return message;
+      case "cancelled":
+        return streamedAgentMessage.status === "created"
+          ? { ...streamedAgentMessage, status: "cancelled" }
```
Comment (PopDaph) [inline front/components/assistant/conversation/AgentMessage.tsx:173]: @spolu I added a commit to change that -> we indeed want streamedAgentMessage but still status cancelled 🙏🏻

**Review by lasryaric — DISMISSED** [review]
LGTM.

File: front/components/assistant/conversation/AgentMessage.tsx (hunk 4 above)
Comment (spolu) [reply front/components/assistant/conversation/AgentMessage.tsx:173]: LGTM

**Review by spolu — APPROVED** [review]

**Pull request merged**